src/frontend/next/src/components/Login/Login.tsx

```diff
@@ -0,0 +1,48 @@
+import Link from 'next/link';
+import { ListItem } from '@material-ui/core';
+import LoggedIn from './LoggedIn';
+import LoggedOut from './LoggedOut';
+import useSiteMetadata from '../../hooks/use-site-metadata';
+import { useUser } from '../UserProvider';
+
+/**
+ * Show either a Login button (if user isn't authenticated)
+ * or a welcome message and Logout button.
+ */
+
+type LoginProps = {
+  // style is a class name passed from MobileHeader component, so it need to be a string
+  style?: string;
+};
+
+const Login = ({ style }: LoginProps) => {
+  const { telescopeUrl } = useSiteMetadata();
+  const user = useUser();
+
+  // Check if using mobile header which means a style prop exists
+  if (style) {
+    return user.isLoggedIn ? (
+      // This is How you make NextJS Link works with ListItem
+      // Reference: https://dev.to/ivandotv/using-next-js-link-component-with-material-ui-buttons-and-menu-items-3m6a
+      <Link href={`${telescopeUrl}/auth/logout`} passHref>
+        <ListItem button component="a" className={style}>
+          <LoggedIn user={user} />
+        </ListItem>
+      </Link>
+    ) : (
+      <Link href={`${telescopeUrl}/auth/login`} passHref>
+        <ListItem button component="a" className={style}>
+          <LoggedOut />
+        </ListItem>
+      </Link>
+    );
+  }
+
+  return user.isLoggedIn ? <LoggedIn user={user} /> : <LoggedOut />;
+};
+
+Login.defaultProps = {
```

Comment:
Let's not bother with this, we'll type props with TS.

Reply:
@humphd :since style is optional and only applied when on mobile. ESLint asked me to have a defaultProps. Should I disabled the lint error or continue with this defaultProps approach?